docs/troubleshooting-guide/datawarehouse.md

```diff
@@ -70,3 +70,49 @@ Fixing this depends on your specific setup. Here are some potentially helpful re
 - [Elastic Load Balancing Connection Timeout Management](https://aws.amazon.com/blogs/aws/elb-idle-timeout-control/)
 - [Heroku timeouts](https://devcenter.heroku.com/articles/request-timeout)
 - [App Engine: Dealing with DeadlineExceededErrors](https://cloud.google.com/appengine/articles/deadlineexceedederrors)
+
+
+### MySQL: Unable to log in with correct credentials
+#### How to detect this
+Metabase fails to connect to your MySQL server with the error message "Looks like the username or password is incorrect", but you are sure that the username and password is correct. You may have created the MySQL user with an allowed host other than that which you are connecting from.
+
+For example, if the MySQL server is running in a Docker container, and your `metabase` user was created with `CREATE USER 'metabase'@'localhost' IDENTIFIED BY 'thepassword';`, the `localhost` will be resolved to the Docker container, and not the host machine, causing access to be denied.
+
+You can identify this issue, by looking in the Metabase server logs for the error message `Access denied for user 'metabase'@'172.17.0.1' (using password: YES)`. Note the host name `172.17.0.1` (in this case a Docker network IP address), and `using password: YES` at the end.
+
+You will see the same error message when attempting to connect to the MySQL server with the command-line client: `mysql -h 127.0.0.1 -u metabase -p`
+
+#### How to fix this
+Recreate the MySQL user with the correct host name: `CREATE USER 'metabase'@'172.17.0.1' IDENTIFIED BY 'thepassword';`. Otherwise, if necessary, a wildcard may be used for the host name: `CREATE USER 'metabase'@'%' IDENTIFIED BY 'thepassword';`
+
+That user's permissions will need to be set:
+
+```sql
+GRANT SELECT ON targetdb.* TO 'metabase'@'172.17.0.1';
+FLUSH PRIVILEGES;
+```
+
+Remember to `DROP USER 'metabase'@'localhost';` the old user.
+
+
+### MySQL: Unable to log in to MySQL 8 with correct credentials
+
+#### How to detect this
+Metabase fails to connect to your MySQL 8 server with the error message "Looks like the username or password is incorrect", and the Metabase server logs include the error message `Access denied for user 'metabase'@'172.17.0.1' (using password: NO)`. Note the `using password: NO` at the end.
+
+You may still be able to successfully connect to the server using another MySQL client, such as the command-line client: `mysql -h 127.0.0.1 -u metabase -p`
+
+#### How to fix this
+Change the authentication plugin used by the Metabase user's to `mysql_native_password`: `ALTER USER 'metabase'@'%' IDENTIFIED WITH mysql_native_password BY 'thepassword';`
```

Comment:
I think there's an unnecessary apostrophe here.

`used by the Metabase user's to` should be `used by the Metabase user to` I think.